crates/nu-cli/src/repl.rs

```diff
@@ -229,22 +270,38 @@ pub fn evaluate_repl(
 
         start_time = std::time::Instant::now();
 
+        line_editor = if config.use_ansi_coloring {
+            line_editor.with_hinter(Box::new({
+                let stack = stack.get_unique_mut();
```

Comment:
this line is why the hinter is moved up: we need to evaluate something in the stack, get the result, and then box it up. Right after this we end up capturing the `stack` and can no longer get RW access to the stack